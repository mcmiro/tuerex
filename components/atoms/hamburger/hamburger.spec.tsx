/* eslint-disable no-use-before-define */
import * as React from "react";
import {screen, render} from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Hamburger from ".";

describe("Test Hamburger component", () => {
	it("Should pass when isActive and ontogglemenu is defined", () => {
		const isActive = false;
		const toggleMock = jest.fn();

		render(<Hamburger isActive={isActive} onToggleMenu={toggleMock} data-testid="hamurger-test" />);

		const hamburger = screen.queryByTestId("hamurger-test");
		expect(hamburger).toBeDefined();
		expect(hamburger).toMatchSnapshot();
	});

	it("Should pass when isActive props is change on true", async () => {
		const isActive = false;
		const toggleMock = jest.fn();

		render(<Hamburger isActive={isActive} onToggleMenu={toggleMock} data-testid="hamurger-test" />);

		const hamburger = screen.queryByTestId("hamurger-test");
		hamburger && userEvent.click(hamburger);
		hamburger && expect(toggleMock).toHaveBeenCalledTimes(1);
		hamburger && expect(isActive).toBe(true);
	});
});
