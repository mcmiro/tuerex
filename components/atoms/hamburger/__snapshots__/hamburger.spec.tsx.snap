// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Hamburger component Should pass when isActive and ontogglemenu is defined 1`] = `null`;
